networkx/linalg/simrank.py

```diff
@@ -0,0 +1,94 @@
+"""SimRank matrix of graphs
+"""
+
+# Copyright (C) 2015 by
+# Janu Verma <j.verma5@gmail.com>
+# All rights reserved.
+# BSD License. 
+
+
+import networkx as nx
+import numpy as np 
+import itertools
+
+
+__author__ = "\n".join(['Janu Verma (j.verma5@gmail.com)'])
+
+__all__ = ['simrank_matrix']
+
+def simrank_matrix(G, relativeImportanceFactor=0.9, max_iter=100, tol=1e-4):
+	"""Return the SimRank matrix of G.
+
+	SimRank is a vertex similarity measure. It computes the 
+	similarity between two nodes on a graph based on the topology, 
+	i.e., the nodes and the links of the graph.
+
+	Parameters
+	----------
+	G : graph
+		A networkx graph
+
+	relativeImportanceFactor : float, optional (default = 0.9)
+		It represents the relative importance between 
+		in-direct neighbors and direct neighbors.
+
+	max_iter : integer, optional (default=100)
+		Maximum number of iterations. 
+
+	tol : float, optional
+		Error tolerance used to check convergence 
+
+	Returns
+	-------
+	simrank : Numpy matrix
+		Matrix containing SimRank scores of the nodes.
+
+	Examples
+	--------
+	>>> G = nx.DiGraph(nx.path_graph(4))
+	>>> sr = nx.simrank_matrix(G)
+
+	Notes
+	-----
+
+	See Also
+	--------
+	modularity_matrix
+	pagerank
+	laplacian_matrix
+
+	References
+	----------
+	.. [1] G. Jeh and J. Widom. "SimRank: a measure of structural-context similarity",
+	 	In KDD'02: Proceedings of the eighth ACM SIGKDD international conference on Knowledge discovery and data mining,
+	 	pages 538-543. ACM Press, 2002
+	"""
+	nodes = G.nodes()
+	n = len(nodes)
+	nodesDict = {node:i for (i,node) in enumerate(nodes)}
```

Comment:
variables should be lowercase. rename `nodesDict` -> `nodes_dict` or `ndict`
same for other variables